Rename theme.spacing to theme.space

This spec seems to be settling in for things like theme-ui, styled-system, and others.

https://system-ui.com/theme/#:~:text=This%20specification%20is%20a%20work%2Din%2Dprogress.&text=The%20theme%20object%20is%20intended,%2C%20and%2For%20design%20tokens.

The key for spacing is just "space".  I think it would be better to use this same key. It would allow Bedrock to be plugged into existing projects that use this spec, and it's even a natural extension point for projects that don't. Or, allow the key to be provided somehow so it's variable.

Hoping you can help me. I am having trouble getting the spacing typing
working right. What is strange is it works on storybook no problem. I have
a key "spacing" on my styled-components theme and it picked it up.

packages/forms-list/src/Components/WelcomeView/WelcomeView.tsx:192:21 -
error TS2769: No overload matches this call.
  Overload 1 of 2, '(props:
Omit<Omit<Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>,
HTMLDivElement>, "key" | keyof HTMLAttributes<...>> & { ...; } &
InlineClusterProps & InlineProps, never> & Partial<...>, "theme"> & { ...;
} & { ...; }): ReactElement<...>', gave the following error.
    Type '"k-space-4-x"' is not assignable to type 'keyof Spacing'.
  Overload 2 of 2, '(props: StyledComponentPropsWithAs<"div", any,
InlineClusterProps & InlineProps, never>):
ReactElement<StyledComponentPropsWithAs<"div", any, InlineClusterProps &
InlineProps, never>, string | JSXElementConstructor<...>>', gave the
following error.
    Type '"k-space-4-x"' is not assignable to type 'keyof Spacing'.

192             <Inline gutter="k-space-4-x" stretch="all">

On Thu, May 20, 2021 at 5:58 PM Pete Smith ***@***.***> wrote:

> you can just default if not provided to what you have today. Just make it
> configurable via the spacing constants package I would assume?
>
> On Thu, May 20, 2021 at 5:57 PM Justin Travis Waith-Mair <
> ***@***.***> wrote:
>
>> That makes a lot of sense. Technically this would be a breaking Change
>> but given that it's still early on we could get away with it
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Bedrock-Layouts/Bedrock/issues/542#issuecomment-845503839>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAJFD66XUEYZETFKOAKL34DTOWAUVANCNFSM45HTBW6Q>
>> .
>>
>
>
> --
> Pete Smith
> klaviyo.com <https://www.klaviyo.com>
> [image: Klaviyo Logo]
>


-- 
Pete Smith
klaviyo.com <https://www.klaviyo.com>
[image: Klaviyo Logo]

-- 
*Any information contained in or attached to this e-mail is intended solely 
for the use of the intended recipient(s) and may contain certain 
information that is confidential, proprietary and/or legally privileged. If 
you are not the intended recipient, you may not review, copy or distribute 
this message. If you receive this in error, please notify the sender and 
destroy all copies of this message and attachments.**
*

If you use one of the original spacings like ‘xl’. Does it work?

Justin Travis Waith-Mair

> On May 20, 2021, at 6:16 PM, Pete Smith ***@***.***> wrote:
> 
> ﻿
> Hoping you can help me. I am having trouble getting the spacing typing
> working right. What is strange is it works on storybook no problem. I have
> a key "spacing" on my styled-components theme and it picked it up.
> 
> packages/forms-list/src/Components/WelcomeView/WelcomeView.tsx:192:21 -
> error TS2769: No overload matches this call.
> Overload 1 of 2, '(props:
> Omit<Omit<Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>,
> HTMLDivElement>, "key" | keyof HTMLAttributes<...>> & { ...; } &
> InlineClusterProps & InlineProps, never> & Partial<...>, "theme"> & { ...;
> } & { ...; }): ReactElement<...>', gave the following error.
> Type '"k-space-4-x"' is not assignable to type 'keyof Spacing'.
> Overload 2 of 2, '(props: StyledComponentPropsWithAs<"div", any,
> InlineClusterProps & InlineProps, never>):
> ReactElement<StyledComponentPropsWithAs<"div", any, InlineClusterProps &
> InlineProps, never>, string | JSXElementConstructor<...>>', gave the
> following error.
> Type '"k-space-4-x"' is not assignable to type 'keyof Spacing'.
> 
> 192 <Inline gutter="k-space-4-x" stretch="all">
> 
> On Thu, May 20, 2021 at 5:58 PM Pete Smith ***@***.***> wrote:
> 
> > you can just default if not provided to what you have today. Just make it
> > configurable via the spacing constants package I would assume?
> >
> > On Thu, May 20, 2021 at 5:57 PM Justin Travis Waith-Mair <
> > ***@***.***> wrote:
> >
> >> That makes a lot of sense. Technically this would be a breaking Change
> >> but given that it's still early on we could get away with it
> >>
> >> —
> >> You are receiving this because you authored the thread.
> >> Reply to this email directly, view it on GitHub
> >> <https://github.com/Bedrock-Layouts/Bedrock/issues/542#issuecomment-845503839>,
> >> or unsubscribe
> >> <https://github.com/notifications/unsubscribe-auth/AAJFD66XUEYZETFKOAKL34DTOWAUVANCNFSM45HTBW6Q>
> >> .
> >>
> >
> >
> > --
> > Pete Smith
> > klaviyo.com <https://www.klaviyo.com>
> > [image: Klaviyo Logo]
> >
> 
> 
> -- 
> Pete Smith
> klaviyo.com <https://www.klaviyo.com>
> [image: Klaviyo Logo]
> 
> -- 
> *Any information contained in or attached to this e-mail is intended solely 
> for the use of the intended recipient(s) and may contain certain 
> information that is confidential, proprietary and/or legally privileged. If 
> you are not the intended recipient, you may not review, copy or distribute 
> this message. If you receive this in error, please notify the sender and 
> destroy all copies of this message and attachments.**
> *
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.

--- a/.storybook/pages/Landing.stories.mdx
+++ b/.storybook/pages/Landing.stories.mdx
@@ -27,7 +27,7 @@ import { Center } from '@bedrock-layout/center'
 
 [![lerna](https://img.shields.io/badge/maintained%20with-lerna-cc00ff.svg)](https://lerna.js.org/)
 [![Commitizen friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg)](http://commitizen.github.io/cz-cli/)
-[![Build Status](https://travis-ci.org/Bedrock-Layouts/Bedrock.svg?branch=master)](https://travis-ci.org/Bedrock-Layouts/Bedrock)
+[![Build Status](https://travis-ci.com/Bedrock-Layouts/Bedrock.svg?branch=master)](https://travis-ci.com/Bedrock-Layouts/Bedrock)
 ![GitHub](https://img.shields.io/github/license/Bedrock-Layouts/Bedrock)
 
 </Center>

--- a/README.md
+++ b/README.md
@@ -4,7 +4,7 @@
 
 [![lerna](https://img.shields.io/badge/maintained%20with-lerna-cc00ff.svg)](https://lerna.js.org/)
 [![Commitizen friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg)](http://commitizen.github.io/cz-cli/)
-[![Build Status](https://travis-ci.org/Bedrock-Layouts/Bedrock.svg?branch=master)](https://travis-ci.org/Bedrock-Layouts/Bedrock)
+[![Build Status](https://travis-ci.com/Bedrock-Layouts/Bedrock.svg?branch=master)](https://travis-ci.com/Bedrock-Layouts/Bedrock)
 ![GitHub](https://img.shields.io/github/license/Bedrock-Layouts/Bedrock)
 ![GitHub Repo stars](https://img.shields.io/github/stars/Bedrock-Layouts/Bedrock?style=social)
 

--- a/packages/appboundary/src/index.tsx
+++ b/packages/appboundary/src/index.tsx
@@ -1,16 +1,13 @@
 import { Center } from "@bedrock-layout/center";
-import { mergeBreakpoints } from "@bedrock-layout/spacing-constants";
 import React from "react";
-import styled, { ThemeContext } from "styled-components";
+import styled from "styled-components";
 
-const safeTheme = { breakPoints: {} };
+const XX_LARGE_BREAKPOINT = 1920;
 
 export const AppBoundary = styled.div.attrs((props) => {
-  const { breakPoints } = React.useContext(ThemeContext) ?? safeTheme;
-  const maxWidth = mergeBreakpoints(breakPoints).xxlarge;
   return {
     "data-bedrock-layout-appboundary": "",
-    children: <Center maxWidth={maxWidth}>{props.children}</Center>,
+    children: <Center maxWidth={XX_LARGE_BREAKPOINT}>{props.children}</Center>,
   };
 })`
   padding: 0;

--- a/packages/center/src/index.tsx
+++ b/packages/center/src/index.tsx
@@ -1,24 +1,19 @@
-import { mergeBreakpoints } from "@bedrock-layout/spacing-constants";
 import PropTypes from "prop-types";
 import styled from "styled-components";
 
+const MEDIUM_BREAKPOINT = "1023px";
 export interface CenterProps {
   maxWidth?: number | string;
   centerText?: boolean;
   centerChildren?: boolean;
 }
 
-function getSafeMaxWidth(
-  breakPoints: Record<string, unknown>,
-  maxWidth?: number | string
-) {
+function getSafeMaxWidth(maxWidth?: number | string) {
   if (typeof maxWidth === "string") {
     return maxWidth;
   }
 
-  return typeof maxWidth === "number"
-    ? `${maxWidth}px`
-    : mergeBreakpoints(breakPoints).medium + "px";
+  return typeof maxWidth === "number" ? `${maxWidth}px` : MEDIUM_BREAKPOINT;
 }
 
 export const Center = styled.div.attrs<CenterProps>(() => {
@@ -29,10 +24,9 @@ export const Center = styled.div.attrs<CenterProps>(() => {
   @property --maxWidth {
     syntax: "<length-percentage>";
     inherits: false;
-    initial-value: 1023px;
+    initial-value: ${MEDIUM_BREAKPOINT};
   }
-  --maxWidth: ${({ maxWidth, theme: { breakPoints } }) =>
-    getSafeMaxWidth(breakPoints, maxWidth)};
+  --maxWidth: ${({ maxWidth }) => getSafeMaxWidth(maxWidth)};
 
   box-sizing: content-box;
 
@@ -40,14 +34,10 @@ export const Center = styled.div.attrs<CenterProps>(() => {
   margin-inline-end: auto;
   margin-inline: auto;
 
-  max-inline-size: 1023px;
+  max-inline-size: ${MEDIUM_BREAKPOINT};
 
-  @supports (
-    max-inline-size:
-      ${({ maxWidth, theme: { breakPoints } }) =>
-        getSafeMaxWidth(breakPoints, maxWidth)}
-  ) {
-    max-inline-size: var(--maxWidth, 1023px);
+  @supports (max-inline-size: ${({ maxWidth }) => getSafeMaxWidth(maxWidth)}) {
+    max-inline-size: var(--maxWidth, ${MEDIUM_BREAKPOINT});
   }
 
   ${(props) => (props.centerText ? "text-align: center;" : "")}

--- a/packages/grid/src/index.tsx
+++ b/packages/grid/src/index.tsx
@@ -1,27 +1,25 @@
 import {
   SpacingOptions,
   getSpacingValue,
-  mergeBreakpoints,
 } from "@bedrock-layout/spacing-constants";
 import PropTypes from "prop-types";
 import styled from "styled-components";
+
+const SMALL_BREAKPOINT = "639px";
 
 export interface GridProps {
   gutter: keyof SpacingOptions;
   minItemWidth?: number | string;
 }
 
-function getSafeMinItemWidth(
-  breakPoints: Record<string, unknown>,
-  minItemWidth?: number | string
-) {
+function getSafeMinItemWidth(minItemWidth?: number | string) {
   if (typeof minItemWidth === "string") {
     return minItemWidth;
   }
 
   return typeof minItemWidth === "number"
     ? `${minItemWidth}px`
-    : mergeBreakpoints(breakPoints).smallOnly + "px";
+    : SMALL_BREAKPOINT;
 }
 
 export const Grid = styled.div.attrs<GridProps>(() => {
@@ -31,24 +29,24 @@ export const Grid = styled.div.attrs<GridProps>(() => {
 })<GridProps>`
   box-sizing: border-box;
   @property --minItemWidth {
-    syntax: "<length>";
+    syntax: "<length-percentage>";
     inherits: false;
-    initial-value: 639px;
+    initial-value: ${SMALL_BREAKPOINT};
   }
 
   --gutter: ${({ gutter, theme }) => getSpacingValue(theme, gutter) ?? "0px"};
-  --minItemWidth: ${({ minItemWidth, theme }) =>
-    getSafeMinItemWidth(theme.breakPoints ?? {}, minItemWidth)};
+  --minItemWidth: ${({ minItemWidth }) => getSafeMinItemWidth(minItemWidth)};
 
   display: grid;
   gap: var(--gutter);
 
-  grid-template-columns: repeat(auto-fit, minmax(min(639, 100%), 1fr));
+  grid-template-columns: repeat(
+    auto-fit,
+    minmax(min(${SMALL_BREAKPOINT}, 100%), 1fr)
+  );
 
   @supports (
-    width:
-      ${({ minItemWidth, theme }) =>
-        getSafeMinItemWidth(theme.breakPoints ?? {}, minItemWidth)}
+    width: ${({ minItemWidth }) => getSafeMinItemWidth(minItemWidth)}
   ) {
     grid-template-columns: repeat(
       auto-fit,

--- a/packages/spacing-constants/examples/spacing-constants.stories.mdx
+++ b/packages/spacing-constants/examples/spacing-constants.stories.mdx
@@ -4,54 +4,76 @@ import { Meta, Story, Canvas, ArgsTable, Preview } from '@storybook/addon-docs/b
 
 # Spacing Constants
 
-Spacing constants are the standard by which all spacing and layout decisions are made in the Bedrock Layout Primitives. Generally speaking, you won't typically need to pull these in to your App Directly since the Layout Primitives already implement them, but these do exist incase you ever need the raw values.
+Spacing constants are the standard by which all spacing and layout decisions are made in the Bedrock Layout Primitives. Generally speaking, you won't typically need to pull these into your App Directly since the Layout Primitives already implement them. Still, these do exist in case you ever need the raw values.
 
-The spacing is built simply on `1rem` as the starting mid point and then doubling the value at each step above and halfing the value each step below. This comes down to the following spacings: `0`,`0.0625`, `0.125rem`, `0.25rem`, `0.5rem`, `1rem`, `2rem`, `4rem`. Each one of these spacing points are then named based on t-shirt size or `none`, `xxs`, `xs`, `sm`, `md`, `lg`, `xl`, and `xxl` respectively. Middle point in between the larger spacings are also added to allow you to finely tune the points, e.g. `mdLg` is `0.75rem`.
+The spacing is built on `1rem` as the starting point. We then double the value at each step above `1rem` and half the value at each step below `1rem`. Each of these spacing points is then given a name based on a t-shirt size like `md`, `lg`, `xl`, and `xxl`. There are also middle point between the larger spacings, e.g. `mdLg`, that allow you to finely tune your spacings.
 
-For the breakpoints, we are adopting the foundation breakpoint values. All values are given in `rems`. There are two types of constants, single value and range values. `medium`, `large`, `xlarge`, and `xxlarge` are intended communicate that size or smaller. The constants are `smallOnly`, `mediumOnly`, `largeOnly`, and `xlargeOnly` and will be an array with the first value being the lowerbound and the second value being the upperbound;.
+## Spacing values
+
+| space   | size        |
+| ------- | ----------- |
+| `none`  | `0`         |
+| `xxs`   | `0.0625rem` |
+| `xs`    | `0.125rem`  |
+| `sm`    | `0.25rem`   |
+| `md`    | `0.5rem`    |
+| `mdLg`  | `0.75rem`   |
+| `lg`    | `1rem`      |
+| `lgXl`  | `1.5rem`    |
+| `xl`    | `2rem`      |
+| `xlXXl` | `3rem`      |
+| `xxl`   | `4rem`      |
 
 ## Usage
 
+These values are already built in to those that deal with spacing, like the `Stack` or `PadBox` components. In the unlikely senario that you do need to access the values directly, you can import the values like this:
+
 ```javascript
-import { spacing, breakpoints } from '@bedrock-layout/spacing-constants';
+import { spacing } from '@bedrock-layout/spacing-constants';
 
 spacing.sm; //0.25rem
 
-breakpoints.large; //64rem
+spacing.xxl; //4rem
 ```
 
-## Overriding the values
+## Integrating With Your Design System
 
-All of the Bedrock Layout Primitives are built using `styled-components`. In order to override the values of any or all of the constants,
-you can use the `ThemeProvider` to change the values. The `ThemeProvider` can be rendered at any level of the tree and will override the
-values at that point. [Check out the docs for `styled-components`](https://styled-components.com/docs/advanced#theming) for more details on
-how it works.
+Bedrock Layout Primitives is designed to be able to be integrated easily into any design system. Bedrock's Spacing values can be overridden using `styled-components` `ThemeProvider`. You can [Check out the docs for `styled-components`](https://styled-components.com/docs/advanced#theming). Bedrock Layout Primitives follow the [System UI Theme Specification](https://system-ui.com/theme/#:~:text=This%20specification%20is%20a%20work%2Din%2Dprogress.&text=The%20theme%20object%20is%20intended,%2C%20and%2For%20design%20tokens) when overriding theme values. To override Bedrock Layout's default space values, you will need to pass in your new spacing values as an object of key/value pairs to the `space` key of the theme. This
 
-```javascript
+```jsx
 import { ThemeProvider } from 'styled-components';
 
 const newSpacings = {
-    sm:'45px',
-    xxl:'100ch'
+    "0x": 0,
+    "1x":'45px',
+    "3x":'100ch'
 }
 
-<ThemeProvider theme={{ spacing: newSpacings }}>
-    <Stack gutter="sm">
+<ThemeProvider theme={{ space: newSpacings }}>
+    <Stack gutter="1x">
      {...}
     </Stack>
 </ThemeProvider>;
 ```
 
-| spacing | breakpoints  |
-| ------- | ------------ |
-| `none`  | `medium`     |
-| `xxs`   | `large`      |
-| `xs`    | `large`      |
-| `sm`    | `xlarge`     |
-| `md`    | `xxlarge`    |
-| `mdLg`  | `smallOnly`  |
-| `lg`    | `mediumOnly` |
-| `lgXl`  | `largeOnly`  |
-| `xl`    | `xlargeOnly` |
-| `xlXXl` | -            |
-| `xxl`   | -            |
+The spacing values can either be a CSS size unit, as a string, or a positive number for the number of pixels
+
+## Overriding Spacing Types in TypeScript
+
+In order to override the types, you must follow one additional step. `styled-components` allows you to [define your theme types through declaration merging](https://styled-components.com/docs/api#create-a-declarations-file). In your declaration file you will need to define your types like this:
+
+```typescript
+import 'styled-components';
+
+declare module 'styled-components' {
+  export interface DefaultTheme {
+    space: {
+      '0x': number;
+      '1x': string;
+      '3x': string;
+    };
+  }
+}
+```
+
+Space values can only be of type `string` or `number`.

--- a/packages/switcher/package.json
+++ b/packages/switcher/package.json
@@ -3,7 +3,7 @@
   "version": "2.1.0",
   "description": "bedrock-layout switcher layout primitive",
   "sideEffects": false,
-  "private": false,
+  "private": true,
   "publishConfig": {
     "access": "public"
   },

